plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1312 @@
+}
+
+// Gather Gathers data for all servers.
+func (h *JBoss) Gather(acc telegraf.Accumulator) error {
+
+	if h.ResponseTimeout.Duration < time.Second {
+		h.ResponseTimeout.Duration = time.Second * 5
+	}
+
+	if h.client.HTTPClient() == nil {
+		tlsCfg, err := h.ClientConfig.TLSConfig()
+
+		if err != nil {
+			return err
+		}
+		tr := &http.Transport{
+			TLSClientConfig: tlsCfg,
+		}
+		client := &http.Client{
+			Transport: tr,
+			Timeout:   h.ResponseTimeout.Duration,
+		}
+		h.client.SetHTTPClient(client)
+	}
+
+	var wg sync.WaitGroup
+	for _, server := range h.Servers {
+
+		//Check Exec Mode for each servers
+
+		bodyContent, err := h.createRequestBody(GetExecStat, nil)
+		if err != nil {
+			acc.AddError(err)
+		}
+
+		out, err := h.doRequest(server, bodyContent)
+		if err != nil {
+			log.Printf("E! JBoss Error handling ExecMode Test: %s\n", err)
+			acc.AddError(err)
+		}
+		// Unmarshal json
+		exec := ExecTypeResponse{}
+		if err = json.Unmarshal(out, &exec); err != nil {
+			acc.AddError(fmt.Errorf("Error decoding JSON response (ExecTypeResponse) %s,%s", out, err))
+			return nil
+		}
+		var execAsDomain bool
+		var isEAP7 bool
+		if exec.Result["launch-type"] == "DOMAIN" {
+			execAsDomain = true
+		} else {
+			execAsDomain = false
+		}
+		isEAP7 = isEAP7Version(exec.Result["product-name"].(string), exec.Result["product-version"].(string))
+		log.Printf("D! JBoss Plugin Working as Domain: %t EAP7: %t  for server %s \n", execAsDomain, isEAP7, server)
+
+		wg.Add(1)
+		go func(server string, execAsDomain bool) {
+			defer wg.Done()
+			//default as standalone server
+			hosts := HostResponse{Outcome: "", Result: []string{"standalone"}}
+
+			if execAsDomain {
+				bodyContent, err := h.createRequestBody(GetHosts, nil)
+				if err != nil {
+					acc.AddError(err)
+				}
+
+				out, err := h.doRequest(server, bodyContent)
+
+				if err != nil {
+					log.Printf("E! JBoss Error handling response 1: %s\n", err)
+					log.Printf("E! JBoss server:%s bodyContent %s\n", server, bodyContent)
+					acc.AddError(err)
+					return
+				}
+				// Unmarshal json
+
+				if err = json.Unmarshal(out, &hosts); err != nil {
+					acc.AddError(fmt.Errorf("Error decoding JSON response (HostResponse) %s :%s", out, err))
+				}
+				log.Printf("D! JBoss HOSTS %s", hosts)
+			}
+
+			h.getServersOnHost(acc, server, execAsDomain, isEAP7, hosts.Result)
+
+		}(server, execAsDomain)
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+// Check if JBoss is EE 6 or EE 7
+// Parameters:
+//     productName   : The tname of the product
+//     productVersion: version of the produkt
+//
+// Returns:
+//     error: Any error that may have occurred
+
+func isEAP7Version(productName string, productVersion string) bool {
+	log.Printf("D! JBoss product: %s version %s", productName, productVersion)
+	if strings.Contains(productName, "EAP") {
+		return strings.HasPrefix(productVersion, "7.")
+	} else {
+		return strings.HasPrefix(productVersion, "10.")
+	}
```

Comment:
```suggestion
	if strings.Contains(productName, "EAP") {
		return strings.HasPrefix(productVersion, "7.")
	}
	return strings.HasPrefix(productVersion, "10.")
```

Reply:
Furthermore, this function is a four-liner only used once. Maybe just move the function body to the respective location and get rid of the function...